test/FinancialSystem-test.ts

```diff
@@ -0,0 +1,298 @@
+                        .connect(acc2)
+                        .issueLoan(0, 200)
+                    ).to.be.revertedWith("Loan amount must be greater than zero");
+                    // duration = 0
+                    expect(
+                        loanSystem
+                        .connect(acc2)
+                        .issueLoan(amountLoan, 0)
+                    ).to.be.revertedWith("Loan duration must be greater than zero");
+                });
+            });
+
+            describe("Test repayLoan", () => {
+                it("Should repayment loan and emit Loan Repaid", async () => {
+                    const {loanSystem, accounts} = await loadFixture(deployLoanSystem);
+                    const acc2 = accounts[1];
+                    const amountLoan = ethers.parseEther("20"); // 20ETH
+                    await loanSystem.deposit({value: ethers.parseEther("100")}); // nap cho contract 100ETH           
+    
+                    // acc2 vay 20 ETH lãi suất 2 và thời gian 100 ngày
+                    await loanSystem.connect(acc2).issueLoan(amountLoan, 100); 
+                    
+                    //vay 20 ETH trả 30 ETH
+                    const amount = ethers.parseEther("30");
+                    await loanSystem.connect(acc2).repayLoan({value: amount});
+
+                    const calculateInterest = await loanSystem.calculateInterest(amountLoan, 100); // tính lãi của 20 ETH
+                    const totalAmount = calculateInterest + amountLoan; // tổng số tiền phải trả
+                    const balance = amount - totalAmount; // số tiền dư sau khi trả xog khoản vay
+                    
+                    const loans = await loanSystem.loans(acc2.address);
+                    const loanAmount = loans.amount;
+                    // console.log(loans);
+                    // số tiền nợ trả về 0
+                    expect(loanAmount).to.equal(0);
+                    
+                    //Số tiền dư phải được cộng vào ví có thể rút được
+                    expect(
+                        await loanSystem.getBalanceDeposist(acc2.address)
+                    ).to.equal(balance);
+                });
+
+                it("should revert insufficient value to repayment loan", async () => {
+                    const {loanSystem, accounts} = await loadFixture(deployLoanSystem);
+                    const acc2 = accounts[1];
+                    const amountLoan = ethers.parseEther("20"); // 20ETH
+                    const value = ethers.parseEther("10"); // 10 ETH
+                    await loanSystem.deposit({value: ethers.parseEther("100")}); // deposit for contract 100ETH           
+    
+                    //If No outstanding loan for this address
+                    expect(
+                        loanSystem
+                        .connect(acc2)
+                        .repayLoan({value: value})
+                    ).to.be.revertedWith("No outstanding loan for this address");
```

Comment:
Hình như chỗ này chưa đúng lắm. Lúc này value = ethers.parseEther("10"); thì sao revert về "No outstanding loan for this address" được?